src/ur_realtime_communication.cpp

```diff
@@ -84,17 +84,25 @@ void UrRealtimeCommunication::halt() {
 	comThread_.join();
 }
 
-void UrRealtimeCommunication::addCommandToQueue(std::string inp) {
+void UrRealtimeCommunication::addCommandToQueue(const std::string& inp) {
 	int bytes_written;
-	if (inp.back() != '\n') {
-		inp.append("\n");
-	}
+  assert( inp.back == '\n');
```

Comment:
Behavior is changed here. Before, a trailing `\n` would be automatically added if not present, now program will be aborted. Any reason for this?